rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/compact/NodeRemover.java

```diff
@@ -0,0 +1,276 @@
+        if (findNodeSource() && ((node.isDeployingNode() && findDeploying()) || (findHost() && findNode()))) {
+            return Optional.of(index);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Tries to find and remove node source with the same id (sourceName) as given
+     * @param nodeSource
+     */
+    void findAndRemove(NodeSource nodeSource) {
+        init(nodeSource);
+        if (findNodeSource()) {
+            hierarchyNodeSourceIterator.remove();
+            compactFlowPanel.remove(index);
+        }
+    }
+
+    /**
+     * Tries to find and remove node with the same id (nodeUrl) as given
+     * @param node
+     */
+    public void findAndRemove(Node node) {
+        init(node);
+        if (findNodeSource()) {
+            if (node.isDeployingNode() && findDeploying()) {
+                removeDeployingNode();
+            } else {
+                if (findHost() && findNode()) {
+                    removeNode();
+
+                    // remove dangling host
+                    if (hierarchyHost.isDangling()) {
+                        removeHost();
+                    }
+
+                    return;
+                }
+            }
+        }
+    }
+
+    protected void removeHost() {
+        hierarchyHostIterator.remove();
+        hierarchyNodeSource.decrementTiles();
+        compactFlowPanel.remove(index - 1);
+    }
+
+    protected void removeNode() {
+        nodeIterator.remove();
+        hierarchyNodeSource.decrementTiles();
+        hierarchyHost.decrementTiles();
+        compactFlowPanel.remove(index);
+    }
+
+    private void removeDeployingNode() {
+        nodeIterator.remove();
+        hierarchyNodeSource.decrementTiles();
+        compactFlowPanel.remove(index);
+    }
+
+    /**
+     * Tries to find node source, if find, it sets fields dedicated to node source
+     * and make index point to this node source.
+     * @return true if node source was found
+     */
+    boolean findNodeSource() {
+        final Iterator<HierarchyNodeSource> nsIterator = compactFlowPanel.getModel().iterator();
+        while (nsIterator.hasNext()) {
+            final HierarchyNodeSource hierarchyNodeSource = nsIterator.next();
+            if (hierarchyNodeSource.getNodeSource().getSourceName().equals(sourceName)) {
+                this.hierarchyNodeSource = hierarchyNodeSource;
+                this.hierarchyNodeSourceIterator = nsIterator;
+                return true;
+            } else {
+                index += hierarchyNodeSource.getTilesNumber();
+            }
+        }
+        return false;
+    }
+
+    /**
+     * SHOULD be called after findNodeSource().
+     * Tries to find deploying node in the node source found before.
+     * @return true if deploying node was found
+     */
+    private boolean findDeploying() {
+        ++index;
+        final Iterator<NodeSource.Host.Node> nodeIterator = hierarchyNodeSource.getDeploying().iterator();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "nodeIterator" which hides the field declared at line 53. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)